pkg/api/controller_test.go

```diff
@@ -1549,3 +1550,150 @@ func TestController_MergeDiffWithParent(t *testing.T) {
 		t.Fatal("Diff results not as expected:", diff)
 	}
 }
+
+func TestController_MergeIntoStage(t *testing.T) {
+	clt, deps := setupClientWithAdmin(t, "")
+	ctx := context.Background()
+	// setup env
+	repo := testUniqueRepoName()
+	_, err := deps.catalog.CreateRepository(ctx, repo, onBlock(deps, repo), "main")
+	testutil.Must(t, err)
+	_, err = deps.catalog.CreateBranch(ctx, repo, "branch1", "main")
+	testutil.Must(t, err)
+	testutil.MustDo(t, "create entry bar1", deps.catalog.CreateEntry(ctx, repo, "branch1", catalog.DBEntry{Path: "foo/bar1", PhysicalAddress: "bar1addr", CreationDate: time.Now(), Size: 1, Checksum: "cksum1"}))
+	commit1, err := deps.catalog.Commit(ctx, repo, "branch1", "some message", DefaultUserID, nil)
+	testutil.Must(t, err)
+
+	// merge 'branch1' to 'main' staging
+	commitMessage := "add foo1"
```

Comment:
Maybe change the commit message so that it is related to the merge operation?

Reply:
Will remove it the auto-generated message from the server is better - same operation as our clients